### PR TITLE
Fixed some todo parsing corner cases.

### PR DESCRIPTION
**Pull request by criske**
Fixes for these corner cases:

```java
 /**
     * Method G.
     *
     *      @todo #200:30min Single line todo method G.
     *  This line is not part of todo even if a space is aligned with start.
     *
     */
    public void methodG() {

    }
```
Current parser would interpret the body as: **_Single line todo method G. line is not part of todo even if a space is aligned with start._** instead of **_Single line todo method G._**.

```java
/**
     * Method H.
     *
     * @todo #200:30min Single line todo method H.
     *  @todo #201:30min This a new todo an not part of the above todo.
     *
     */
    public void methodG() {

    }
```
Current parser would interpret the second todo as line body for the above one.

---
No bounty for this, it was a quick fix and most of the code is refactoring.

### CODE REVIEW

**Comment by charlesmike** [comment]
@criske thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by charlesmike** [comment]
@amihaiemil I couldn't find any assignee for this task. This is either because there are no contributors with role ``REV`` available or because the project does not have enough funds.

Please, make sure there is at least one available contributor with the required role and the project can afford to pay them.

**Comment by criske** [comment]
btw, about this case
```java
/**
     * Method H.
     *
     * @todo #200:30min First line todo method H,
     *  this is the 2nd line
     *       is the 3rd line part of the todo?
     *  
     *
     */
    public void methodH() {

    }
```
  is the 3rd line part of the todo? Right now the parser interprets it as part of todo.

**Comment by amihaiemil** [comment]
@criske 

>   is the 3rd line part of the todo? Right now the parser interprets it as part of todo.

Well, I think it's debatable. If you ask me, it's ok to be part of the todo. So let's leave it as it is for now. 
Thanks!

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by amihaiemil** [comment]
@criske looks like a test is failing on Travis, though :( 

Can you have a look pls? :D

**Comment by criske** [comment]
@amihaiemil That's the test that fails only on ci https://github.com/self-xdsd/todo-finder-cli/pull/73#discussion_r553857495 . I'll try to edit the test file, maybe it'll fix it.

File: src/test/java/com/selfxdsd/todocli/JsonTodosSerializerTestCase.java [hunk 1]
```diff
@@ -57,7 +57,12 @@ public void shouldSerializeTodos() throws IOException {
         final URI location = serializer.serialize();
         final JsonReader reader = Json
             .createReader(location.toURL().openStream());
-        final JsonArray json = reader.readArray();
+        final JsonArray json = Json.createPatchBuilder()
+            .replace("/1/id", -2091730235)
+            .replace("/1/author", "cristianpela")
+            .replace("/1/timestamp", "2021-01-18 11:32:28 +0200")
+            .build()
+            .apply(reader.readArray());
```
Comment (criske) [inline src/test/java/com/selfxdsd/todocli/JsonTodosSerializerTestCase.java:65]: @amihaiemil I came up with this workaround...  I can't fix that test, maybe travis.yml has some way to invalidate the cache or something.

**Comment by criske** [comment]
It looks like it's all fixed, the other todo in the test file needed an update too.

**Comment by amihaiemil** [comment]
@criske thank you!

**Pull request merged**